Xamarin.Forms.Platform.iOS/DisposeHelpers.cs

```diff
@@ -0,0 +1,52 @@
+namespace Xamarin.Forms.Platform.iOS
+{
+	internal static class DisposeHelpers
+	{
+		internal static void DisposeModalAndChildRenderers(this Element view)
+		{
+			IVisualElementRenderer renderer;
+
+			foreach (VisualElement child in view.Descendants())
+			{
+				renderer = Platform.GetRenderer(child);
+				child.ClearValue(Platform.RendererProperty);
+
+				if (renderer != null)
+				{
+					renderer.NativeView.RemoveFromSuperview();
+					renderer.Dispose();
+				}
+			}
+
+			renderer = Platform.GetRenderer((VisualElement)view);
+			if (renderer != null)
+			{
+				if (renderer.ViewController?.ParentViewController is ModalWrapper modalWrapper)
+					modalWrapper.Dispose();
+
+				renderer.NativeView.RemoveFromSuperview();
+				renderer.Dispose();
+			}
+			view.ClearValue(Platform.RendererProperty);
+		}
+
+		internal static void DisposeRendererAndChildren(this IVisualElementRenderer rendererToRemove)
+		{
+			if (rendererToRemove == null)
```

Comment:
does this really happen ? as it's an extension method, it's like calling (null as IVER).DisposeRendererAndChildren();

it works, but whoever does that deserves a good old NRE instead of returning null